spyderlib/spyder.py

```diff
@@ -2653,6 +2659,65 @@ def start_open_files_server(self):
                 self.sig_open_external_file.emit(fname)
             req.sendall(b' ')
 
+    # ---- Quit and restart
+    def restart(self):
+        """ """
+        # Get start path to use in restart script
+        spyder_start_directory = osp.dirname(osp.abspath(__file__))
+        spyder_folder = osp.split(spyder_start_directory)[0]
+        restart_script = osp.join(spyder_start_directory, 'spyder_restart.py')
+
+        # Get any initial argument passed when spyder was started
+        # Variables defined in bootstrap.py and spyderlib\start_app.py
+        env = os.environ.copy()
+        bootstrap_args = env.pop('SPYDER_BOOTSTRAP_ARGS', None)
+        spyder_args = env.pop('SPYDER_ARGS')
+
+        # Get current process and python running spyder
+        pid = os.getpid()
+        python = sys.executable
+        #python = python.lower().replace('pythonw.exe', 'python.exe')
+
+        # Check if started with bootstrap.py
+        is_bootstrap = False
+        if bootstrap_args is not None:
+            spyder_args = bootstrap_args
+            is_bootstrap = True
+
+        # Pass args as environment variables to restarter subprocess
+        env['SPYDER_ARGS'] = spyder_args
+        env['SPYDER_PID'] = str(pid)
+        env['SPYDER_IS_BOOTSTRAP'] = str(is_bootstrap)
+        env['SPYDER_FOLDER'] = spyder_folder
+
+        # Build the command depending on the OS
+        import platform
+        system = platform.system().lower()
+
+        flags = 0  # Default value
+
+        startupinfo = None
+        if 'win' in system:
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+            command = '"{0}" "{1}"'
+        elif 'linux' in system:
+            command = '"{0}" "{1}"'
+        elif 'darwin' in system or 'osx' in system:
+            command = '"{0}" "{1}"'

```

Comment:
I suppose that you left it for debugging, but `command` is the same in all 3 cases.